app/helpers/application_helper.rb

```diff
@@ -134,35 +134,39 @@ def icon_tag(type, options = {})
     content_tag :i, '', options.merge(class: css_classes)
   end
 
-  def link_to_delete(path, options = {})
-    text = options[:text] || 'Delete'
-    disabled_reason = options[:disabled]
-    if disabled_reason
-      content_tag :span, text, title: disabled_reason, class: 'mouseover'
+  def link_to_delete(
+    path, text: 'Delete', question: nil, type_to_delete: false, remove_container: false, disabled: false, **options
+  )
+    if disabled
+      content_tag :span, text, title: disabled, class: 'mouseover'
     else
-      resource = Array(path).last
       message =
-        if question = options.delete(:question)
+        if question
           question
-        elsif resource.is_a?(ActiveRecord::Base)
+        elsif path.is_a?(ActiveRecord::Base)
+          resource = path
+          name, path = link_parts_for_resource(resource)
+          "Delete #{resource.class.name.split("::").last} #{name} ?"
+        elsif (resource = Array(path).last) && resource.is_a?(ActiveRecord::Base)
           "Delete this #{resource.class.name.split("::").last} ?"
         else
-          "Are you sure ?"
+          "Really delete ?"
         end
-      options[:data] = {confirm: message, method: :delete}
-      if container = options[:remove_container]
-        options[:data][:remove_container] = container
+      options[:data] = {method: :delete}
+      if type_to_delete
+        options[:data][:type_to_delete] = message
+      else
+        options[:data][:confirm] = message
+      end
+      if remove_container
+        options[:data][:remove_container] = remove_container
```

Comment:
set to `true` instead?

Reply:
keeping it open for future enhancements, someone might set something else, afaik no need to preemptively break that